chore: use `FLOAT64_MAX_SAFE_NTH_DOUBLE_FACTORIAL`

anandkaranubc opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

- Completes the TODO on replacing the magic number with `FLOAT64_MAX_SAFE_NTH_DOUBLE_FACTORIAL` 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
chore: use FLOAT64_MAX_SAFE_NTH_DOUBLE_FACTORIAL

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (dismissed):
Thanks, @anandkaranubc!

anandkaranubc commented:
> Thanks, @anandkaranubc!

Thanks for the review, @Planeshifter!

One quick question, I think we need to change the value of `FLOAT64_MAX_SAFE_NTH_DOUBLE_FACTORIAL` from 301 to 300..?

```
In [18]: base.factorial2(301) <= FLOAT64_MAX
Out[18]: false

In [19]: base.factorial2(300) <= FLOAT64_MAX
Out[19]: true
```

https://github.com/stdlib-js/stdlib/blob/a77fb9eba6a9649654a8913e431501fde6bf04da/lib/node_modules/%40stdlib/constants/float64/max-safe-nth-double-factorial/lib/index.js#L44

[Reference](https://www.wolframalpha.com/input?i=double+factorial+301): `factorial2(301) ≈ 1.13 × 10^309`

```
In [34]: FLOAT64_MAX
Out[34]: 1.7976931348623157e+308
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/factorial2/lib/main.js:
```diff
@@ -63,7 +59,7 @@ function factorial2( n ) {
 		return NaN;
 	}
-	if ( n >= MAX_FACTORIAL2 ) {
+	if ( n >= FLOAT64_MAX_SAFE_NTH_DOUBLE_FACTORIAL ) {
```
```suggestion
	if ( n > FLOAT64_MAX_SAFE_NTH_DOUBLE_FACTORIAL ) {
```

Which would also mean changing this, after the constant has been corrected.

Planeshifter commented:
@anandkaranubc I believe you are correct. 

cc @gunjjoshi @kgryte, as this constant was added in https://github.com/stdlib-js/stdlib/pull/2966/files. Maybe the inlined definition of `MAX_FACTORIAL2` here was the source of the value of `301`, or how was the maximum double factorial safely representable within IEEE-754 double-precision floating-point limits determined?

anandkaranubc committed:
fix: inequality check for divergence

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
The changes have been made according to the discussion in the OH.

cc: @Planeshifter @kgryte
